[network] Binding throws java.lang.OutOfMemoryError: Java heap space on start

System:
- CubieTruck running Debian Stretch
- using ezdk-1.8.0_152-8.25.0.76-eval-linux_aarch32hf
- two network cards:
  - eth0: 10.x.y.z (built-in)
  - eth1: 192.168.x.y (USB)

Changes in runtime/bin/setenv:
- export JAVA_PERM_MEM=512M
- export JAVA_MAX_PERM_MEM=1024M
- export JAVA_HOME="/opt/ezdk/"

Other infos:
- Error appeared when switching from 2.1.0 to 2.2.0.
- Also reproducible with latest SNAPSHOT 1189.
- When restarting system it takes up to 12 minutes to start OH2 which causes rules to not be triggered anymore.

How to reproduce:
- make a clean manual install under /opt/openhab2
- try to install Network Binding via PaperUI
- it takes a few minutes with 100% CPU and logs following error under userdata/logs/openhab.log:

```
content of openhab.log:
--- 8< ------ 8< ------ 8< ---
2018-01-19 20:08:36.931 [ERROR] [org.openhab.binding.network         ] - [org.openhab.binding.network.internal.discovery.NetworkDiscoveryService(181)] Error during instantiation of the implementation object
java.lang.OutOfMemoryError: Java heap space
        at java.lang.AbstractStringBuilder.<init>(AbstractStringBuilder.java:68) ~[?:?]
        at java.lang.StringBuilder.<init>(StringBuilder.java:89) ~[?:?]
        at org.apache.commons.net.util.SubnetUtils.format(SubnetUtils.java:284) ~[?:?]
        at org.apache.commons.net.util.SubnetUtils.access$600(SubnetUtils.java:28) ~[?:?]
        at org.apache.commons.net.util.SubnetUtils$SubnetInfo.getAllAddresses(SubnetUtils.java:190) ~[?:?]
        at org.openhab.binding.network.internal.utils.NetworkUtils.getNetworkIPs(NetworkUtils.java:130) ~[?:?]
        at org.openhab.binding.network.internal.utils.NetworkUtils.getNetworkIPs(NetworkUtils.java:113) ~[?:?]
        at org.openhab.binding.network.internal.discovery.NetworkDiscoveryService.<init>(NetworkDiscoveryService.java:67) ~[?:?]
        at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method) ~[?:?]
        at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62) ~[?:?]
        at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45) ~[?:?]
        at java.lang.reflect.Constructor.newInstance(Constructor.java:423) ~[?:?]
        at java.lang.Class.newInstance(Class.java:442) ~[?:?]
        at org.apache.felix.scr.impl.manager.SingleComponentManager.createImplementationObject(SingleComponentManager.java:237) ~[?:?]
        at org.apache.felix.scr.impl.manager.SingleComponentManager.createComponent(SingleComponentManager.java:109) ~[?:?]
        at org.apache.felix.scr.impl.manager.SingleComponentManager.getService(SingleComponentManager.java:906) ~[?:?]
        at org.apache.felix.scr.impl.manager.SingleComponentManager.getServiceInternal(SingleComponentManager.java:879) ~[?:?]
        at org.apache.felix.scr.impl.manager.SingleComponentManager.getService(SingleComponentManager.java:823) ~[?:?]
        at org.eclipse.osgi.internal.serviceregistry.ServiceFactoryUse$1.run(ServiceFactoryUse.java:212) ~[?:?]
        at java.security.AccessController.doPrivileged(Native Method) ~[?:?]
        at org.eclipse.osgi.internal.serviceregistry.ServiceFactoryUse.factoryGetService(ServiceFactoryUse.java:210) ~[?:?]
        at org.eclipse.osgi.internal.serviceregistry.ServiceFactoryUse.getService(ServiceFactoryUse.java:111) ~[?:?]
        at org.eclipse.osgi.internal.serviceregistry.ServiceConsumer$2.getService(ServiceConsumer.java:45) ~[?:?]
        at org.eclipse.osgi.internal.serviceregistry.ServiceRegistrationImpl.getService(ServiceRegistrationImpl.java:496) ~[?:?]
        at org.eclipse.osgi.internal.serviceregistry.ServiceRegistry.getService(ServiceRegistry.java:461) ~[?:?]
        at org.eclipse.osgi.internal.framework.BundleContextImpl.getService(BundleContextImpl.java:619) ~[?:?]
        at com.eclipsesource.jaxrs.publisher.internal.ResourceTracker.addingService(ResourceTracker.java:39) ~[?:?]
        at org.osgi.util.tracker.ServiceTracker$Tracked.customizerAdding(ServiceTracker.java:941) ~[?:?]
        at org.osgi.util.tracker.ServiceTracker$Tracked.customizerAdding(ServiceTracker.java:870) ~[?:?]
        at org.osgi.util.tracker.AbstractTracked.trackAdding(AbstractTracked.java:256) ~[?:?]
        at org.osgi.util.tracker.AbstractTracked.track(AbstractTracked.java:229) ~[?:?]
        at org.osgi.util.tracker.ServiceTracker$Tracked.serviceChanged(ServiceTracker.java:901) ~[?:?]
2018-01-19 20:08:37.208 [WARN ] [org.openhab.binding.network         ] - FrameworkEvent WARNING - org.openhab.binding.network
org.osgi.framework.ServiceException: org.apache.felix.scr.impl.manager.SingleComponentManager.getService() returned a null service object
        at org.eclipse.osgi.internal.serviceregistry.ServiceFactoryUse.factoryGetService(ServiceFactoryUse.java:232) ~[?:?]
        at org.eclipse.osgi.internal.serviceregistry.ServiceFactoryUse.getService(ServiceFactoryUse.java:111) ~[?:?]
        at org.eclipse.osgi.internal.serviceregistry.ServiceConsumer$2.getService(ServiceConsumer.java:45) ~[?:?]
        at org.eclipse.osgi.internal.serviceregistry.ServiceRegistrationImpl.getService(ServiceRegistrationImpl.java:496) ~[?:?]
        at org.eclipse.osgi.internal.serviceregistry.ServiceRegistry.getService(ServiceRegistry.java:461) ~[?:?]
        at org.eclipse.osgi.internal.framework.BundleContextImpl.getService(BundleContextImpl.java:619) ~[?:?]
        at com.eclipsesource.jaxrs.publisher.internal.ResourceTracker.addingService(ResourceTracker.java:39) ~[?:?]
        at org.osgi.util.tracker.ServiceTracker$Tracked.customizerAdding(ServiceTracker.java:941) ~[?:?]
        at org.osgi.util.tracker.ServiceTracker$Tracked.customizerAdding(ServiceTracker.java:870) ~[?:?]
        at org.osgi.util.tracker.AbstractTracked.trackAdding(AbstractTracked.java:256) ~[?:?]
        at org.osgi.util.tracker.AbstractTracked.track(AbstractTracked.java:229) ~[?:?]
        at org.osgi.util.tracker.ServiceTracker$Tracked.serviceChanged(ServiceTracker.java:901) ~[?:?]
        at org.eclipse.osgi.internal.serviceregistry.FilteredServiceListener.serviceChanged(FilteredServiceListener.java:109) ~[?:?]
        at org.eclipse.osgi.internal.framework.BundleContextImpl.dispatchEvent(BundleContextImpl.java:915) ~[?:?]
        at org.eclipse.osgi.framework.eventmgr.EventManager.dispatchEvent(EventManager.java:230) ~[?:?]
        at org.eclipse.osgi.framework.eventmgr.ListenerQueue.dispatchEventSynchronous(ListenerQueue.java:148) ~[?:?]
        at org.eclipse.osgi.internal.serviceregistry.ServiceRegistry.publishServiceEventPrivileged(ServiceRegistry.java:862) ~[?:?]
        at org.eclipse.osgi.internal.serviceregistry.ServiceRegistry.publishServiceEvent(ServiceRegistry.java:801) ~[?:?]
        at org.eclipse.osgi.internal.serviceregistry.ServiceRegistrationImpl.register(ServiceRegistrationImpl.java:127) ~[?:?]
        at org.eclipse.osgi.internal.serviceregistry.ServiceRegistry.registerService(ServiceRegistry.java:225) ~[?:?]
        at org.eclipse.osgi.internal.framework.BundleContextImpl.registerService(BundleContextImpl.java:464) ~[?:?]
        at org.apache.felix.scr.impl.manager.AbstractComponentManager$3.register(AbstractComponentManager.java:887) ~[?:?]
        at org.apache.felix.scr.impl.manager.AbstractComponentManager$3.register(AbstractComponentManager.java:874) ~[?:?]
        at org.apache.felix.scr.impl.manager.RegistrationManager.changeRegistration(RegistrationManager.java:132) ~[?:?]
        at org.apache.felix.scr.impl.manager.AbstractComponentManager.registerService(AbstractComponentManager.java:941) ~[?:?]
        at org.apache.felix.scr.impl.manager.AbstractComponentManager.activateInternal(AbstractComponentManager.java:741) ~[?:?]
        at org.apache.felix.scr.impl.manager.AbstractComponentManager.enableInternal(AbstractComponentManager.java:675) ~[?:?]
        at org.apache.felix.scr.impl.manager.AbstractComponentManager.enable(AbstractComponentManager.java:430) ~[?:?]
        at org.apache.felix.scr.impl.manager.ConfigurableComponentHolder.enableComponents(ConfigurableComponentHolder.java:657) ~[?:?]
        at org.apache.felix.scr.impl.BundleComponentActivator.initialEnable(BundleComponentActivator.java:341) ~[?:?]
        at org.apache.felix.scr.impl.Activator.loadComponents(Activator.java:390) ~[?:?]
        at org.apache.felix.scr.impl.Activator.access$200(Activator.java:54) ~[?:?]
        at org.apache.felix.scr.impl.Activator$ScrExtension.start(Activator.java:265) ~[?:?]
        at org.apache.felix.utils.extender.AbstractExtender.createExtension(AbstractExtender.java:254) ~[?:?]
        at org.apache.felix.utils.extender.AbstractExtender.modifiedBundle(AbstractExtender.java:227) ~[?:?]
        at org.apache.felix.utils.extender.AbstractExtender.addingBundle(AbstractExtender.java:187) ~[?:?]
        at org.osgi.util.tracker.BundleTracker$Tracked.customizerAdding(BundleTracker.java:469) ~[?:?]
        at org.osgi.util.tracker.BundleTracker$Tracked.customizerAdding(BundleTracker.java:415) ~[?:?]
        at org.osgi.util.tracker.AbstractTracked.trackAdding(AbstractTracked.java:256) ~[?:?]
        at org.osgi.util.tracker.AbstractTracked.track(AbstractTracked.java:229) ~[?:?]
        at org.osgi.util.tracker.BundleTracker$Tracked.bundleChanged(BundleTracker.java:444) ~[?:?]
        at org.eclipse.osgi.internal.framework.BundleContextImpl.dispatchEvent(BundleContextImpl.java:903) ~[?:?]
        at org.eclipse.osgi.framework.eventmgr.EventManager.dispatchEvent(EventManager.java:230) ~[?:?]
        at org.eclipse.osgi.framework.eventmgr.ListenerQueue.dispatchEventSynchronous(ListenerQueue.java:148) ~[?:?]
        at org.eclipse.osgi.internal.framework.EquinoxEventPublisher.publishBundleEventPrivileged(EquinoxEventPublisher.java:213) ~[?:?]
        at org.eclipse.osgi.internal.framework.EquinoxEventPublisher.publishBundleEvent(EquinoxEventPublisher.java:120) ~[?:?]
        at org.eclipse.osgi.internal.framework.EquinoxEventPublisher.publishBundleEvent(EquinoxEventPublisher.java:112) ~[?:?]
        at org.eclipse.osgi.internal.framework.EquinoxContainerAdaptor.publishModuleEvent(EquinoxContainerAdaptor.java:156) ~[?:?]
        at org.eclipse.osgi.container.Module.publishEvent(Module.java:476) ~[?:?]
        at org.eclipse.osgi.container.Module.start(Module.java:467) ~[?:?]
        at org.eclipse.osgi.framework.util.SecureAction.start(SecureAction.java:470) ~[?:?]
        at org.eclipse.osgi.container.ModuleContainer.start(ModuleContainer.java:736) ~[?:?]
        at org.eclipse.osgi.container.ModuleContainer.applyDelta(ModuleContainer.java:727) ~[?:?]
        at org.eclipse.osgi.container.ModuleContainer.resolveAndApply(ModuleContainer.java:497) ~[?:?]
        at org.eclipse.osgi.container.ModuleContainer.resolve(ModuleContainer.java:443) ~[?:?]
        at org.eclipse.osgi.container.ModuleContainer.resolve(ModuleContainer.java:433) ~[?:?]
        at org.eclipse.osgi.container.ModuleContainer$ContainerWiring.resolveBundles(ModuleContainer.java:1299) ~[?:?]
        at org.apache.karaf.features.internal.service.FeaturesServiceImpl.resolveBundles(FeaturesServiceImpl.java:1441) ~[?:?]
        at org.apache.karaf.features.internal.service.Deployer.deploy(Deployer.java:874) ~[?:?]
        at org.apache.karaf.features.internal.service.FeaturesServiceImpl.doProvision(FeaturesServiceImpl.java:1248) ~[?:?]
        at org.apache.karaf.features.internal.service.FeaturesServiceImpl.lambda$doProvisionInThread$1(FeaturesServiceImpl.java:1147) ~[?:?]
        at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:?]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:?]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:?]
        at java.lang.Thread.run(Thread.java:748) [?:?]
2018-01-19 20:12:07.251 [ERROR] [org.openhab.binding.network         ] - [org.openhab.binding.network.internal.discovery.NetworkDiscoveryService(181)] Error during instantiation of the implementation object
java.lang.OutOfMemoryError: Java heap space
        at org.apache.commons.net.util.SubnetUtils.format(SubnetUtils.java:284) ~[?:?]
        at org.apache.commons.net.util.SubnetUtils.access$600(SubnetUtils.java:28) ~[?:?]
        at org.apache.commons.net.util.SubnetUtils$SubnetInfo.getAllAddresses(SubnetUtils.java:190) ~[?:?]
        at org.openhab.binding.network.internal.utils.NetworkUtils.getNetworkIPs(NetworkUtils.java:130) ~[?:?]
        at org.openhab.binding.network.internal.utils.NetworkUtils.getNetworkIPs(NetworkUtils.java:113) ~[?:?]
        at org.openhab.binding.network.internal.discovery.NetworkDiscoveryService.<init>(NetworkDiscoveryService.java:67) ~[?:?]
        at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method) ~[?:?]
        at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62) ~[?:?]
        at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45) ~[?:?]
        at java.lang.reflect.Constructor.newInstance(Constructor.java:423) ~[?:?]
        at java.lang.Class.newInstance(Class.java:442) ~[?:?]
        at org.apache.felix.scr.impl.manager.SingleComponentManager.createImplementationObject(SingleComponentManager.java:237) ~[?:?]
        at org.apache.felix.scr.impl.manager.SingleComponentManager.createComponent(SingleComponentManager.java:109) ~[?:?]
        at org.apache.felix.scr.impl.manager.SingleComponentManager.getService(SingleComponentManager.java:906) ~[?:?]
        at org.apache.felix.scr.impl.manager.SingleComponentManager.getServiceInternal(SingleComponentManager.java:879) ~[?:?]
        at org.apache.felix.scr.impl.manager.SingleComponentManager.getService(SingleComponentManager.java:823) ~[?:?]
        at org.eclipse.osgi.internal.serviceregistry.ServiceFactoryUse$1.run(ServiceFactoryUse.java:212) ~[?:?]
        at java.security.AccessController.doPrivileged(Native Method) ~[?:?]
        at org.eclipse.osgi.internal.serviceregistry.ServiceFactoryUse.factoryGetService(ServiceFactoryUse.java:210) ~[?:?]
        at org.eclipse.osgi.internal.serviceregistry.ServiceFactoryUse.getService(ServiceFactoryUse.java:111) ~[?:?]
        at org.eclipse.osgi.internal.serviceregistry.ServiceConsumer$2.getService(ServiceConsumer.java:45) ~[?:?]
        at org.eclipse.osgi.internal.serviceregistry.ServiceRegistrationImpl.getService(ServiceRegistrationImpl.java:496) ~[?:?]
        at org.eclipse.osgi.internal.serviceregistry.ServiceRegistry.getService(ServiceRegistry.java:461) ~[?:?]
        at org.eclipse.osgi.internal.framework.BundleContextImpl.getService(BundleContextImpl.java:619) ~[?:?]
        at org.apache.felix.scr.impl.manager.SingleRefPair.getServiceObject(SingleRefPair.java:72) ~[?:?]
        at org.apache.felix.scr.impl.inject.BindMethod.getServiceObject(BindMethod.java:651) ~[?:?]
        at org.apache.felix.scr.impl.manager.DependencyManager.getServiceObject(DependencyManager.java:2201) ~[?:?]
        at org.apache.felix.scr.impl.manager.DependencyManager.doInvokeBindMethod(DependencyManager.java:1714) ~[?:?]
        at org.apache.felix.scr.impl.manager.DependencyManager.invokeBindMethod(DependencyManager.java:1699) ~[?:?]
        at org.apache.felix.scr.impl.manager.SingleComponentManager.invokeBindMethod(SingleComponentManager.java:372) ~[?:?]
        at org.apache.felix.scr.impl.manager.DependencyManager$MultipleDynamicCustomizer.addedService(DependencyManager.java:320) ~[?:?]
        at org.apache.felix.scr.impl.manager.DependencyManager$MultipleDynamicCustomizer.addedService(DependencyManager.java:291) ~[?:?]
2018-01-19 20:12:07.489 [WARN ] [org.openhab.binding.network         ] - FrameworkEvent WARNING - org.openhab.binding.network
org.osgi.framework.ServiceException: org.apache.felix.scr.impl.manager.SingleComponentManager.getService() returned a null service object
        at org.eclipse.osgi.internal.serviceregistry.ServiceFactoryUse.factoryGetService(ServiceFactoryUse.java:232) ~[?:?]
        at org.eclipse.osgi.internal.serviceregistry.ServiceFactoryUse.getService(ServiceFactoryUse.java:111) ~[?:?]
        at org.eclipse.osgi.internal.serviceregistry.ServiceConsumer$2.getService(ServiceConsumer.java:45) ~[?:?]
        at org.eclipse.osgi.internal.serviceregistry.ServiceRegistrationImpl.getService(ServiceRegistrationImpl.java:496) ~[?:?]
        at org.eclipse.osgi.internal.serviceregistry.ServiceRegistry.getService(ServiceRegistry.java:461) ~[?:?]
        at org.eclipse.osgi.internal.framework.BundleContextImpl.getService(BundleContextImpl.java:619) ~[?:?]
        at org.apache.felix.scr.impl.manager.SingleRefPair.getServiceObject(SingleRefPair.java:72) ~[?:?]
        at org.apache.felix.scr.impl.inject.BindMethod.getServiceObject(BindMethod.java:651) ~[?:?]
        at org.apache.felix.scr.impl.manager.DependencyManager.getServiceObject(DependencyManager.java:2201) ~[?:?]
        at org.apache.felix.scr.impl.manager.DependencyManager.doInvokeBindMethod(DependencyManager.java:1714) ~[?:?]
        at org.apache.felix.scr.impl.manager.DependencyManager.invokeBindMethod(DependencyManager.java:1699) ~[?:?]
        at org.apache.felix.scr.impl.manager.SingleComponentManager.invokeBindMethod(SingleComponentManager.java:372) ~[?:?]
        at org.apache.felix.scr.impl.manager.DependencyManager$MultipleDynamicCustomizer.addedService(DependencyManager.java:320) ~[?:?]
        at org.apache.felix.scr.impl.manager.DependencyManager$MultipleDynamicCustomizer.addedService(DependencyManager.java:291) ~[?:?]
        at org.apache.felix.scr.impl.manager.ServiceTracker$Tracked.customizerAdded(ServiceTracker.java:1215) ~[?:?]
        at org.apache.felix.scr.impl.manager.ServiceTracker$Tracked.customizerAdded(ServiceTracker.java:1136) ~[?:?]
        at org.apache.felix.scr.impl.manager.ServiceTracker$AbstractTracked.trackAdding(ServiceTracker.java:945) ~[?:?]
        at org.apache.felix.scr.impl.manager.ServiceTracker$AbstractTracked.track(ServiceTracker.java:881) ~[?:?]
        at org.apache.felix.scr.impl.manager.ServiceTracker$Tracked.serviceChanged(ServiceTracker.java:1167) ~[?:?]
        at org.apache.felix.scr.impl.BundleComponentActivator$ListenerInfo.serviceChanged(BundleComponentActivator.java:127) ~[?:?]
        at org.eclipse.osgi.internal.serviceregistry.FilteredServiceListener.serviceChanged(FilteredServiceListener.java:109) ~[?:?]
        at org.eclipse.osgi.internal.framework.BundleContextImpl.dispatchEvent(BundleContextImpl.java:915) ~[?:?]
        at org.eclipse.osgi.framework.eventmgr.EventManager.dispatchEvent(EventManager.java:230) ~[?:?]
        at org.eclipse.osgi.framework.eventmgr.ListenerQueue.dispatchEventSynchronous(ListenerQueue.java:148) ~[?:?]
        at org.eclipse.osgi.internal.serviceregistry.ServiceRegistry.publishServiceEventPrivileged(ServiceRegistry.java:862) ~[?:?]
        at org.eclipse.osgi.internal.serviceregistry.ServiceRegistry.publishServiceEvent(ServiceRegistry.java:801) ~[?:?]
        at org.eclipse.osgi.internal.serviceregistry.ServiceRegistrationImpl.register(ServiceRegistrationImpl.java:127) ~[?:?]
        at org.eclipse.osgi.internal.serviceregistry.ServiceRegistry.registerService(ServiceRegistry.java:225) ~[?:?]
        at org.eclipse.osgi.internal.framework.BundleContextImpl.registerService(BundleContextImpl.java:464) ~[?:?]
        at org.apache.felix.scr.impl.manager.AbstractComponentManager$3.register(AbstractComponentManager.java:887) ~[?:?]
        at org.apache.felix.scr.impl.manager.AbstractComponentManager$3.register(AbstractComponentManager.java:874) ~[?:?]
        at org.apache.felix.scr.impl.manager.RegistrationManager.changeRegistration(RegistrationManager.java:132) ~[?:?]
        at org.apache.felix.scr.impl.manager.AbstractComponentManager.registerService(AbstractComponentManager.java:941) ~[?:?]
        at org.apache.felix.scr.impl.manager.AbstractComponentManager.activateInternal(AbstractComponentManager.java:741) ~[?:?]
        at org.apache.felix.scr.impl.manager.AbstractComponentManager.enableInternal(AbstractComponentManager.java:675) ~[?:?]
        at org.apache.felix.scr.impl.manager.AbstractComponentManager.enable(AbstractComponentManager.java:430) ~[?:?]
        at org.apache.felix.scr.impl.manager.ConfigurableComponentHolder.enableComponents(ConfigurableComponentHolder.java:657) ~[?:?]
        at org.apache.felix.scr.impl.BundleComponentActivator.initialEnable(BundleComponentActivator.java:341) ~[?:?]
        at org.apache.felix.scr.impl.Activator.loadComponents(Activator.java:390) ~[?:?]
        at org.apache.felix.scr.impl.Activator.access$200(Activator.java:54) ~[?:?]
        at org.apache.felix.scr.impl.Activator$ScrExtension.start(Activator.java:265) ~[?:?]
        at org.apache.felix.utils.extender.AbstractExtender.createExtension(AbstractExtender.java:254) ~[?:?]
        at org.apache.felix.utils.extender.AbstractExtender.modifiedBundle(AbstractExtender.java:227) ~[?:?]
        at org.apache.felix.utils.extender.AbstractExtender.addingBundle(AbstractExtender.java:187) ~[?:?]
        at org.osgi.util.tracker.BundleTracker$Tracked.customizerAdding(BundleTracker.java:469) ~[?:?]
        at org.osgi.util.tracker.BundleTracker$Tracked.customizerAdding(BundleTracker.java:415) ~[?:?]
        at org.osgi.util.tracker.AbstractTracked.trackAdding(AbstractTracked.java:256) ~[?:?]
        at org.osgi.util.tracker.AbstractTracked.track(AbstractTracked.java:229) ~[?:?]
        at org.osgi.util.tracker.BundleTracker$Tracked.bundleChanged(BundleTracker.java:444) ~[?:?]
        at org.eclipse.osgi.internal.framework.BundleContextImpl.dispatchEvent(BundleContextImpl.java:903) ~[?:?]
        at org.eclipse.osgi.framework.eventmgr.EventManager.dispatchEvent(EventManager.java:230) ~[?:?]
        at org.eclipse.osgi.framework.eventmgr.ListenerQueue.dispatchEventSynchronous(ListenerQueue.java:148) ~[?:?]
        at org.eclipse.osgi.internal.framework.EquinoxEventPublisher.publishBundleEventPrivileged(EquinoxEventPublisher.java:213) ~[?:?]
        at org.eclipse.osgi.internal.framework.EquinoxEventPublisher.publishBundleEvent(EquinoxEventPublisher.java:120) ~[?:?]
        at org.eclipse.osgi.internal.framework.EquinoxEventPublisher.publishBundleEvent(EquinoxEventPublisher.java:112) ~[?:?]
        at org.eclipse.osgi.internal.framework.EquinoxContainerAdaptor.publishModuleEvent(EquinoxContainerAdaptor.java:156) ~[?:?]
        at org.eclipse.osgi.container.Module.publishEvent(Module.java:476) ~[?:?]
        at org.eclipse.osgi.container.Module.start(Module.java:467) ~[?:?]
        at org.eclipse.osgi.framework.util.SecureAction.start(SecureAction.java:470) ~[?:?]
        at org.eclipse.osgi.container.ModuleContainer.start(ModuleContainer.java:736) ~[?:?]
        at org.eclipse.osgi.container.ModuleContainer.applyDelta(ModuleContainer.java:727) ~[?:?]
        at org.eclipse.osgi.container.ModuleContainer.resolveAndApply(ModuleContainer.java:497) ~[?:?]
        at org.eclipse.osgi.container.ModuleContainer.resolve(ModuleContainer.java:443) ~[?:?]
        at org.eclipse.osgi.container.ModuleContainer.resolve(ModuleContainer.java:433) ~[?:?]
        at org.eclipse.osgi.container.ModuleContainer$ContainerWiring.resolveBundles(ModuleContainer.java:1299) ~[?:?]
        at org.apache.karaf.features.internal.service.FeaturesServiceImpl.resolveBundles(FeaturesServiceImpl.java:1441) ~[?:?]
        at org.apache.karaf.features.internal.service.Deployer.deploy(Deployer.java:874) ~[?:?]
        at org.apache.karaf.features.internal.service.FeaturesServiceImpl.doProvision(FeaturesServiceImpl.java:1248) ~[?:?]
        at org.apache.karaf.features.internal.service.FeaturesServiceImpl.lambda$doProvisionInThread$1(FeaturesServiceImpl.java:1147) ~[?:?]
        at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:?]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:?]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:?]
        at java.lang.Thread.run(Thread.java:748) [?:?]
2018-01-19 20:15:54.082 [ERROR] [org.openhab.binding.network         ] - [org.openhab.binding.network.internal.discovery.NetworkDiscoveryService(181)] Error during instantiation of the implementation object
java.lang.OutOfMemoryError: Java heap space
        at java.lang.AbstractStringBuilder.<init>(AbstractStringBuilder.java:68) ~[?:?]
        at java.lang.StringBuilder.<init>(StringBuilder.java:89) ~[?:?]
        at org.apache.commons.net.util.SubnetUtils.format(SubnetUtils.java:284) ~[?:?]
        at org.apache.commons.net.util.SubnetUtils.access$600(SubnetUtils.java:28) ~[?:?]
        at org.apache.commons.net.util.SubnetUtils$SubnetInfo.getAllAddresses(SubnetUtils.java:190) ~[?:?]
        at org.openhab.binding.network.internal.utils.NetworkUtils.getNetworkIPs(NetworkUtils.java:130) ~[?:?]
        at org.openhab.binding.network.internal.utils.NetworkUtils.getNetworkIPs(NetworkUtils.java:113) ~[?:?]
        at org.openhab.binding.network.internal.discovery.NetworkDiscoveryService.<init>(NetworkDiscoveryService.java:67) ~[?:?]
        at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method) ~[?:?]
        at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62) ~[?:?]
        at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45) ~[?:?]
        at java.lang.reflect.Constructor.newInstance(Constructor.java:423) ~[?:?]
        at java.lang.Class.newInstance(Class.java:442) ~[?:?]
        at org.apache.felix.scr.impl.manager.SingleComponentManager.createImplementationObject(SingleComponentManager.java:237) ~[?:?]
        at org.apache.felix.scr.impl.manager.SingleComponentManager.createComponent(SingleComponentManager.java:109) ~[?:?]
        at org.apache.felix.scr.impl.manager.SingleComponentManager.getService(SingleComponentManager.java:906) ~[?:?]
        at org.apache.felix.scr.impl.manager.SingleComponentManager.getServiceInternal(SingleComponentManager.java:879) ~[?:?]
        at org.apache.felix.scr.impl.manager.AbstractComponentManager.activateInternal(AbstractComponentManager.java:749) ~[?:?]
        at org.apache.felix.scr.impl.manager.AbstractComponentManager.enableInternal(AbstractComponentManager.java:675) ~[?:?]
        at org.apache.felix.scr.impl.manager.AbstractComponentManager.enable(AbstractComponentManager.java:430) ~[?:?]
        at org.apache.felix.scr.impl.manager.ConfigurableComponentHolder.enableComponents(ConfigurableComponentHolder.java:657) ~[?:?]
        at org.apache.felix.scr.impl.BundleComponentActivator.initialEnable(BundleComponentActivator.java:341) ~[?:?]
        at org.apache.felix.scr.impl.Activator.loadComponents(Activator.java:390) ~[?:?]
        at org.apache.felix.scr.impl.Activator.access$200(Activator.java:54) ~[?:?]
        at org.apache.felix.scr.impl.Activator$ScrExtension.start(Activator.java:265) ~[?:?]
        at org.apache.felix.utils.extender.AbstractExtender.createExtension(AbstractExtender.java:254) ~[?:?]
        at org.apache.felix.utils.extender.AbstractExtender.modifiedBundle(AbstractExtender.java:227) ~[?:?]
        at org.apache.felix.utils.extender.AbstractExtender.addingBundle(AbstractExtender.java:187) ~[?:?]
        at org.osgi.util.tracker.BundleTracker$Tracked.customizerAdding(BundleTracker.java:469) ~[?:?]
        at org.osgi.util.tracker.BundleTracker$Tracked.customizerAdding(BundleTracker.java:415) ~[?:?]
        at org.osgi.util.tracker.AbstractTracked.trackAdding(AbstractTracked.java:256) ~[?:?]
        at org.osgi.util.tracker.AbstractTracked.track(AbstractTracked.java:229) ~[?:?]


--- 8< ------ 8< ------ 8< ---
```
- edit: added network card info

--- a/addons/binding/org.openhab.binding.network/META-INF/MANIFEST.MF
+++ b/addons/binding/org.openhab.binding.network/META-INF/MANIFEST.MF
@@ -16,6 +16,7 @@ Import-Package:
  org.eclipse.smarthome.config.discovery,
  org.eclipse.smarthome.core.cache,
  org.eclipse.smarthome.core.library.types,
+ org.eclipse.smarthome.core.net,
  org.eclipse.smarthome.core.thing,
  org.eclipse.smarthome.core.thing.binding,
  org.eclipse.smarthome.core.types,

--- a/addons/binding/org.openhab.binding.network/src/main/java/org/openhab/binding/network/internal/utils/NetworkUtils.java
+++ b/addons/binding/org.openhab.binding.network/src/main/java/org/openhab/binding/network/internal/utils/NetworkUtils.java
@@ -33,6 +33,8 @@ import java.util.HashSet;
 import java.util.Iterator;
 import java.util.LinkedHashSet;
 import java.util.Set;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
 
 import org.apache.commons.lang.StringUtils;
 import org.apache.commons.lang.SystemUtils;
@@ -51,6 +53,8 @@ import org.slf4j.LoggerFactory;
 @NonNullByDefault
 public class NetworkUtils {
     private final Logger logger = LoggerFactory.getLogger(NetworkUtils.class);
+    private static final Pattern IP4_ADDRESS = Pattern
+            .compile("^(\\d{1,3}\\.\\d{1,3}\\.\\d{1,3}\\.\\d{1,3})\\/(\\d+)$");
 
     /**
      * Gets every IPv4 Address on each Interface except the loopback
@@ -136,27 +140,48 @@ public class NetworkUtils {
     public Set<String> getNetworkIPs(Set<String> interfaceIPs, int maximumPerInterface) {
         LinkedHashSet<String> networkIPs = new LinkedHashSet<>();
 
-        logger.info("{} interface IPs found: {}", interfaceIPs.size(), interfaceIPs);
+        long minCIDRsuffix = 8; // historic Class A network, addresses = 16777214
+        if (maximumPerInterface != 0) {
+            // calculate minimum CIDR suffix from maximumPerInterface (equals leading unset bits (Integer has 32 bits)
+            minCIDRsuffix = Integer.numberOfLeadingZeros(maximumPerInterface);
+            // if only the highest is set, increase suffix by 1 to cover all addresses
+            if (Integer.bitCount(maximumPerInterface) == 1) {
+                minCIDRsuffix++;
+            }
+        }
+        logger.trace("set minCIDRsuffix to {}, maximumPerInterFace is {}", minCIDRsuffix, maximumPerInterface);
 
         for (String string : interfaceIPs) {
-            logger.info("processing IP: {}", string);
-            try {
-                // gets every ip which can be assigned on the given network
-                SubnetUtils utils = new SubnetUtils(string);
-                logger.info("resolved to network: {}", utils.getInfo());
-                String[] addresses = utils.getInfo().getAllAddresses();
-                int len = addresses.length;
-                if (maximumPerInterface != 0 && maximumPerInterface < len) {
-                    len = maximumPerInterface;
+            Matcher addressMatcher = IP4_ADDRESS.matcher(string);
+            if (addressMatcher.matches()) {
+                if (Integer.valueOf(addressMatcher.group(2)) < minCIDRsuffix) {
+                    logger.info(
+                            "CIDR suffix is smaller than /{} on interface with address {}, truncating to /{}, some addresses might be lost",
+                            minCIDRsuffix, string, minCIDRsuffix);
+                    string = addressMatcher.group(1) + "/" + String.valueOf(minCIDRsuffix);
                 }
-                for (int i = 0; i < len; i++) {
-                    networkIPs.add(addresses[i]);
+                try {
+                    // gets every ip which can be assigned on the given network
+                    SubnetUtils utils = new SubnetUtils(string);
+                    // TODO: remove logging
+                    logger.info("resolved to network: {}", utils.getInfo());
+                    String[] addresses = utils.getInfo().getAllAddresses();
+                    int len = addresses.length;
+                    if (maximumPerInterface != 0 && maximumPerInterface < len) {
+                        len = maximumPerInterface;
+                    }
+                    for (int i = 0; i < len; i++) {
+                        networkIPs.add(addresses[i]);
+                    }
+                } catch (Exception ex) {
                 }
-            } catch (Exception ex) {
+            } else {
+                logger.info("skipping interface with address {}: not an IPv4 address", string);
             }
         }
 
         return networkIPs;
+
     }
 
     /**

--- a/addons/binding/org.openhab.binding.network/src/main/java/org/openhab/binding/network/internal/utils/NetworkUtils.java
+++ b/addons/binding/org.openhab.binding.network/src/main/java/org/openhab/binding/network/internal/utils/NetworkUtils.java
@@ -163,8 +163,6 @@ public class NetworkUtils {
                 try {
                     // gets every ip which can be assigned on the given network
                     SubnetUtils utils = new SubnetUtils(string);
-                    // TODO: remove logging
-                    logger.info("resolved to network: {}", utils.getInfo());
                     String[] addresses = utils.getInfo().getAllAddresses();
                     int len = addresses.length;
                     if (maximumPerInterface != 0 && maximumPerInterface < len) {

--- a/addons/binding/org.openhab.binding.network/src/main/java/org/openhab/binding/network/internal/utils/NetworkUtils.java
+++ b/addons/binding/org.openhab.binding.network/src/main/java/org/openhab/binding/network/internal/utils/NetworkUtils.java
@@ -18,9 +18,9 @@ import java.io.InputStreamReader;
 import java.net.ConnectException;
 import java.net.DatagramPacket;
 import java.net.DatagramSocket;
+import java.net.Inet4Address;
 import java.net.InetAddress;
 import java.net.InetSocketAddress;
-import java.net.InterfaceAddress;
 import java.net.NetworkInterface;
 import java.net.NoRouteToHostException;
 import java.net.PortUnreachableException;
@@ -30,17 +30,18 @@ import java.net.SocketException;
 import java.net.SocketTimeoutException;
 import java.util.Enumeration;
 import java.util.HashSet;
-import java.util.Iterator;
 import java.util.LinkedHashSet;
 import java.util.Set;
-import java.util.regex.Matcher;
 import java.util.regex.Pattern;
+import java.util.stream.Collectors;
 
 import org.apache.commons.lang.StringUtils;
 import org.apache.commons.lang.SystemUtils;
 import org.apache.commons.net.util.SubnetUtils;
 import org.eclipse.jdt.annotation.NonNullByDefault;
 import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.smarthome.core.net.CidrAddress;
+import org.eclipse.smarthome.core.net.NetUtil;
 import org.eclipse.smarthome.io.net.exec.ExecUtil;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
@@ -62,39 +63,9 @@ public class NetworkUtils {
      *
      * @return The collected IPv4 Addresses
      */
-    public Set<String> getInterfaceIPs() {
-        Set<String> interfaceIPs = new HashSet<>();
-
-        Enumeration<NetworkInterface> interfaces;
-        try {
-            interfaces = NetworkInterface.getNetworkInterfaces();
-        } catch (SocketException ignored) {
-            // If we are not allowed to enumerate, we return an empty result set.
-            return interfaceIPs;
-        }
-
-        // For each interface ...
-        for (Enumeration<NetworkInterface> en = interfaces; en.hasMoreElements();) {
-            NetworkInterface networkInterface = en.nextElement();
-            boolean isLoopback = true;
-            try {
-                isLoopback = networkInterface.isLoopback();
-            } catch (SocketException ignored) {
-            }
-            if (!isLoopback) {
-                // .. and for each address ...
-                for (Iterator<InterfaceAddress> it = networkInterface.getInterfaceAddresses().iterator(); it
-                        .hasNext();) {
-
-                    // ... get IP and Subnet
-                    InterfaceAddress interfaceAddress = it.next();
-                    interfaceIPs.add(interfaceAddress.getAddress().getHostAddress() + "/"
-                            + interfaceAddress.getNetworkPrefixLength());
-                }
-            }
-        }
-
-        return interfaceIPs;
+    public Set<CidrAddress> getInterfaceIPs() {
+        return NetUtil.getAllInterfaceAddresses().stream().filter(a -> a.getAddress() instanceof Inet4Address)
+                .collect(Collectors.toSet());
     }
 
     /**
@@ -137,49 +108,42 @@ public class NetworkUtils {
      * @param maximumPerInterface The maximum of IP addresses per interface or 0 to get all.
      * @return Every single IP which can be assigned on the Networks the computer is connected to
      */
-    public Set<String> getNetworkIPs(Set<String> interfaceIPs, int maximumPerInterface) {
+    public Set<String> getNetworkIPs(Set<CidrAddress> interfaceIPs, int maximumPerInterface) {
         LinkedHashSet<String> networkIPs = new LinkedHashSet<>();
 
-        long minCIDRsuffix = 8; // historic Class A network, addresses = 16777214
+        short minCidrPrefixLength = 8; // historic Class A network, addresses = 16777214
         if (maximumPerInterface != 0) {
-            // calculate minimum CIDR suffix from maximumPerInterface (equals leading unset bits (Integer has 32 bits)
-            minCIDRsuffix = Integer.numberOfLeadingZeros(maximumPerInterface);
-            // if only the highest is set, increase suffix by 1 to cover all addresses
+            // calculate minimum CIDR prefix length from maximumPerInterface
+            // (equals leading unset bits (Integer has 32 bits)
+            minCidrPrefixLength = (short) Integer.numberOfLeadingZeros(maximumPerInterface);
             if (Integer.bitCount(maximumPerInterface) == 1) {
-                minCIDRsuffix++;
+                // if only the highest is set, decrease prefix by 1 to cover all addresses
+                minCidrPrefixLength--;
             }
         }
-        logger.trace("set minCIDRsuffix to {}, maximumPerInterFace is {}", minCIDRsuffix, maximumPerInterface);
+        logger.trace("set minCidrPrefixLength to {}, maximumPerInterface is {}", minCidrPrefixLength,
+                maximumPerInterface);
 
-        for (String string : interfaceIPs) {
-            Matcher addressMatcher = IP4_ADDRESS.matcher(string);
-            if (addressMatcher.matches()) {
-                if (Integer.valueOf(addressMatcher.group(2)) < minCIDRsuffix) {
-                    logger.info(
-                            "CIDR suffix is smaller than /{} on interface with address {}, truncating to /{}, some addresses might be lost",
-                            minCIDRsuffix, string, minCIDRsuffix);
-                    string = addressMatcher.group(1) + "/" + String.valueOf(minCIDRsuffix);
-                }
-                try {
-                    // gets every ip which can be assigned on the given network
-                    SubnetUtils utils = new SubnetUtils(string);
-                    String[] addresses = utils.getInfo().getAllAddresses();
-                    int len = addresses.length;
-                    if (maximumPerInterface != 0 && maximumPerInterface < len) {
-                        len = maximumPerInterface;
-                    }
-                    for (int i = 0; i < len; i++) {
-                        networkIPs.add(addresses[i]);
-                    }
-                } catch (Exception ex) {
-                }
-            } else {
-                logger.info("skipping interface with address {}: not an IPv4 address", string);
+        for (CidrAddress cidrNotation : interfaceIPs) {
+            if (cidrNotation.getPrefix() < minCidrPrefixLength) {
+                logger.info(
+                        "CIDR prefix is smaller than /{} on interface with address {}, truncating to /{}, some addresses might be lost",
+                        minCidrPrefixLength, cidrNotation, minCidrPrefixLength);
+                cidrNotation = new CidrAddress(cidrNotation.getAddress(), minCidrPrefixLength);
+            }
+
+            SubnetUtils utils = new SubnetUtils(cidrNotation.toString());
+            String[] addresses = utils.getInfo().getAllAddresses();
+            int len = addresses.length;
+            if (maximumPerInterface != 0 && maximumPerInterface < len) {
+                len = maximumPerInterface;
+            }
+            for (int i = 0; i < len; i++) {
+                networkIPs.add(addresses[i]);
             }
         }
 
         return networkIPs;
-
     }
 
     /**

--- a/addons/binding/org.openhab.binding.network/src/main/java/org/openhab/binding/network/internal/utils/NetworkUtils.java
+++ b/addons/binding/org.openhab.binding.network/src/main/java/org/openhab/binding/network/internal/utils/NetworkUtils.java
@@ -32,7 +32,6 @@ import java.util.Enumeration;
 import java.util.HashSet;
 import java.util.LinkedHashSet;
 import java.util.Set;
-import java.util.regex.Pattern;
 import java.util.stream.Collectors;
 
 import org.apache.commons.lang.StringUtils;
@@ -54,8 +53,6 @@ import org.slf4j.LoggerFactory;
 @NonNullByDefault
 public class NetworkUtils {
     private final Logger logger = LoggerFactory.getLogger(NetworkUtils.class);
-    private static final Pattern IP4_ADDRESS = Pattern
-            .compile("^(\\d{1,3}\\.\\d{1,3}\\.\\d{1,3}\\.\\d{1,3})\\/(\\d+)$");
 
     /**
      * Gets every IPv4 Address on each Interface except the loopback

--- a/addons/binding/org.openhab.binding.network/src/main/java/org/openhab/binding/network/internal/utils/NetworkUtils.java
+++ b/addons/binding/org.openhab.binding.network/src/main/java/org/openhab/binding/network/internal/utils/NetworkUtils.java
@@ -40,6 +40,8 @@ import org.apache.commons.net.util.SubnetUtils;
 import org.eclipse.jdt.annotation.NonNullByDefault;
 import org.eclipse.jdt.annotation.Nullable;
 import org.eclipse.smarthome.io.net.exec.ExecUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 /**
  * Network utility functions for pinging and for determining all interfaces and assigned IP addresses.
@@ -48,6 +50,8 @@ import org.eclipse.smarthome.io.net.exec.ExecUtil;
  */
 @NonNullByDefault
 public class NetworkUtils {
+    private final Logger logger = LoggerFactory.getLogger(NetworkUtils.class);
+
     /**
      * Gets every IPv4 Address on each Interface except the loopback
      * The Address format is ip/subnet
@@ -132,10 +136,14 @@ public class NetworkUtils {
     public Set<String> getNetworkIPs(Set<String> interfaceIPs, int maximumPerInterface) {
         LinkedHashSet<String> networkIPs = new LinkedHashSet<>();
 
+        logger.info("{} interface IPs found: {}", interfaceIPs.size(), interfaceIPs);
+
         for (String string : interfaceIPs) {
+            logger.info("processing IP: {}", string);
             try {
                 // gets every ip which can be assigned on the given network
                 SubnetUtils utils = new SubnetUtils(string);
+                logger.info("resolved to network: {}", utils.getInfo());
                 String[] addresses = utils.getInfo().getAllAddresses();
                 int len = addresses.length;
                 if (maximumPerInterface != 0 && maximumPerInterface < len) {

